src/app/components/audio-recordings/pages/analysis-results/analyses-results.component.ts

```diff
@@ -0,0 +1,185 @@
+const regionKey = "region";
+const siteKey = "site";
+
+export const rootPath = "/analysis_jobs/system/results/";
+
+@Component({
+  selector: "baw-analyses-results",
+  templateUrl: "analyses-results.component.html",
+  styleUrls: ["analyses-results.component.scss"],
+})
+export class AnalysesResultsComponent extends PageComponent implements OnInit {
+  public constructor(
+    public api: AnalysisJobItemResultsService,
+    private route: ActivatedRoute
+  ) {
+    super();
+  }
+
+  public rows$: Observable<AnalysisJobItemResult[]>;
+  private readonly routeData = this.route.snapshot.data;
+  private readonly audioRecording: AudioRecording = this.routeData[audioRecordingKey]?.model;
+  private rows = Array<AnalysisJobItemResult>();
+
+  public get analysisJob() {
+    // since this component is currently only used for default analysis jobs, we can hard code the attributes of this job
+    return new AnalysisJob({
+      name: "system",
+      id: "system",
+    });
+  }
+
+  public ngOnInit() {
+    this.getItems()
+      .then((items: AnalysisJobItemResult[]) => {
+        this.rows = items;
+        this.updateRows();
+      });
+  }
+
+  public updateRows = (): Observable<AnalysisJobItemResult[]> =>
+    this.rows$ = this.getRows();
+
+  public getRows() {
+    const data = new Observable<AnalysisJobItemResult[]>((observer) => {
+      observer.next(this.rows);
+    });
+    return data;
+  }
+
+  /**
+   * Fetches the `AnalysisJobItemResult` model from the baw-api
+   *
+   * @param model The id of the model that is requested from the baw-api
+   * @returns The complete `AnalysisJobItemResult` model of the requested item. If the item is not defined, the root path will be returned
+   */
+  public async getItems(model?: AnalysisJobItemResult): Promise<AnalysisJobItemResult[]> {
+    return new Promise((resolveTo) => {
+      const analysisJobId = this.analysisJob;
+      const response = lastValueFrom(this.api
+        .list(analysisJobId, this.audioRecording.id, model)
+        .pipe(first())
+        .pipe(
+          takeUntil(this.unsubscribe)
+        )
+      );
+
+      resolveTo(response);
+    });
+  }
+
+  // this method needs to be refactored
+  public toggleRow(model: AnalysisJobItemResult) {
+    this.getModelChildren(model.children ?? [], model).then((childItems) => {
+      this.addParentInformation(childItems, model);
+
+      if (model.open) {
+        // the relative path for all sub folders should include the folders name & two back slashes before and after the name
+        // this prevents false positives e.g. if the folder name is contained within another folders name
+        this.rows = this.rows.filter(item =>
+          // only the root item will not have a parent item, in this case, we can quickly determine that all folders need to be closed
+          !isInstantiated(item.parentItem) ||
+          !item.parentItem.path.includes(`/${model.name}/`) ||
+          item.name === model.name
```

Comment:
You are able to have multiple files open at the same time, but only the root folder will not have a `parentItem`. So therefore if the `parentItem` attribute does not exist on the clicked on item, all filter conditions will fail, causing all folders to collapse.

The second condition removes all items that are sub items of the directory that was clicked on.

The third condition is needed so that the original file is preserved (since the `includes`) condition will also match against the directory that the user clicked on.

Reply:
I'll be changing the functionality in a future commit, because I do agree that these chained conditions can be hard to keep track of